File: code/python/1476.py
#https://leetcode.com/problems/subrectangle-queries/

class SubrectangleQueries:
    def __init__(self, rectangle: List[List[int]]):
        self.data = rectangle

    def updateSubrectangle(self, row1: int, col1: int, row2: int, col2: int, newValue: int) -> None:
        for i in range(col1, col2+1):
            for j in range(row1, row2+1):
               self. data[j][i] = newValue

    def getValue(self, row: int, col: int) -> int:
        return self.data[row][col]

File: code/cpp/1476.cpp
//https://leetcode.com/problems/subrectangle-queries/

class SubrectangleQueries {
private:
    vector<vector<int>> data;
public:
    SubrectangleQueries(vector<vector<int>>& rectangle) {
        data = rectangle;
    }

    void updateSubrectangle(int row1, int col1, int row2, int col2, int newValue) {
        for (size_t i = row1; i <= row2; i++)
            for (size_t j = col1; j <= col2; j++)
                data[i][j] = newValue;
    }

    int getValue(int row, int col) {
        return data[row][col];
    }
};
